### PR TITLE
Sync doc updates for proxyRole to versioned docs

### PR DESCRIPTION
**Pull request by momo-jun**
The code PR https://github.com/apache/pulsar/pull/19455 has been applied to corresponding patch versions of Pulsar.

So this PR applies incremental doc updates to the 2.8.x/2.9.x/2.10.x/2.11.x docs (original doc PR: [#408](https://github.com/apache/pulsar-site/pull/408)).

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: versioned_docs/version-2.10.x/security-authorization.md [hunk 1]
```diff
@@ -38,6 +38,12 @@ By default, the broker treats the connection between a proxy and the broker as a
 
 Pulsar uses *Proxy roles* to enable the authentication. Proxy roles are specified in the broker configuration file, [`conf/broker.conf`](reference-configuration.md#broker). If a client that is authenticated with a broker is one of its ```proxyRoles```, all requests from that client must also carry information about the role of the client that is authenticated with the proxy. This information is called the *original principal*. If the *original principal* is absent, the client is not able to access anything.
 
+:::note
+
+Starting from 2.10.4, if a Proxy is not correctly configured to use a role that is in the `proxyRoles`, the connection will get rejected.
```
Comment (michaeljmarshall) [inline versioned_docs/version-2.10.x/security-authorization.md:43]: This requirement was only introduced in 3.0.0. It isn't present for any prior versions.

**Review by michaeljmarshall — COMMENTED** [review]
Thanks for your contribution @momo-jun. The protocol doc is valid, but we don't need to copy the `note`.

File: versioned_docs/version-2.10.x/security-authorization.md (hunk 1 above)
Comment (momo-jun) [reply versioned_docs/version-2.10.x/security-authorization.md:43]: Oh, I thought the changes were all applied to those patch version branches. I will revert them.

**Review by Anonymitaet — APPROVED** [review]

**Review by michaeljmarshall — APPROVED** [review]
LGTM

**Pull request merged**